dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.AudioToText.cs

```diff
@@ -53,13 +59,34 @@ internal async Task<IReadOnlyList<TextContent>> GetTextFromAudioContentsAsync(
     private static AudioTranscriptionOptions AudioOptionsFromExecutionSettings(OpenAIAudioToTextExecutionSettings executionSettings)
         => new()
         {
-            TimestampGranularities = AudioTimestampGranularities.Default,
+            TimestampGranularities = ConvertTimestampGranularities(executionSettings.TimestampGranularities),
             Language = executionSettings.Language,
             Prompt = executionSettings.Prompt,
             Temperature = executionSettings.Temperature,
             ResponseFormat = ConvertResponseFormat(executionSettings.ResponseFormat)
         };
 
+    private static AudioTimestampGranularities ConvertTimestampGranularities(ICollection<string>? timestampGranularities)
+    {
+        AudioTimestampGranularities result = AudioTimestampGranularities.Default;
+        if (timestampGranularities is null || timestampGranularities.Count == 0)
+        {
+            return result;
+        }
+
+        if (timestampGranularities.Any(granularity => string.Equals(nameof(AudioTimestampGranularities.Word), granularity, StringComparison.OrdinalIgnoreCase)))
+        {
+            result |= AudioTimestampGranularities.Word;
+        }
+
+        if (timestampGranularities.Any(granularity => string.Equals(nameof(AudioTimestampGranularities.Segment), granularity, StringComparison.OrdinalIgnoreCase)))
+        {
+            result |= AudioTimestampGranularities.Segment;
+        }
```

Comment:
nit: This will iterate over `timestampGranularities` two times, while we can achieve the same result iterating just once:
```suggestion
        foreach (var granularity in timestampGranularities)
        {
            if (string.Equals(nameof(AudioTimestampGranularities.Word), granularity, StringComparison.OrdinalIgnoreCase))
            {
                result |= AudioTimestampGranularities.Word;
            }
            
            if (string.Equals(nameof(AudioTimestampGranularities.Segment), granularity, StringComparison.OrdinalIgnoreCase))
            {
                result |= AudioTimestampGranularities.Segment;
            }
        }
```

Reply:
Good observation, added the continue after the first check as well. Thanks! addressed.